src/utils/dimensions.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Handle horizontal and vertical dimensions.
+ *
+ * Notes:
+ * (1) Definition: _verticalResolution = componentHeight / timeInDisplay_
+ *   - e.g. `screenHeight / hoursInDisplay`
+ *   - (time can be in minutes, hours, etc)
+ *
+ * Given a point in the screen, the relation between
+ * its top position (pixels) and its time in the day is:
+ *     top = verticalResolution * time + CONTENT_TOP_PADDING
+ *
+ * When the agenda does not begin at 00:00, use:
+ *     top = verticalResolution * (time - beginAgendaAt) + CONTENT_TOP_PADDING
+ *
+ *
+ * (2) Some functions __must__ be declared as worklets because they
+ * are used with react-native-reanimated in useAnimatedStyle(),
+ * useDerivedValue(), etc.
+ */
+export const HEADER_HEIGHT = 50;
+export const CONTENT_TOP_PADDING = 16;
+
+export const computeVerticalDimensions = (
+  totalHeight,
+  hoursInDisplay,
+  minutesStep,
+) => {
+  const minutesInDisplay = hoursInDisplay * 60;
+  const timeLabelsInDisplay = Math.ceil(minutesInDisplay / minutesStep);
+
+  const minutesResolution = totalHeight / minutesInDisplay;
+
+  return {
+    timeLabelHeight: totalHeight / timeLabelsInDisplay,
+    resolution: minutesResolution,
+  };
+};
+
+/**
+ * Convert time in the day (expressed in minutes) to top (pixels in y dim).
+ *
+ * @param {Number} minutes Minutes to convert
+ * @param {Number} verticalResolution resolution in minutes
+ * @param {Number} minutesOffset offset, e.g. beginAgendaAt
+ * @returns pixels
+ */
+export const minutesInDayToTop = (
+  minutes,
+  verticalResolution,
+  minutesOffset = 0,
+) => {
+  'worklet';
+
+  return (
+    (minutes - (minutesOffset || 0)) * verticalResolution + CONTENT_TOP_PADDING
+  );
+};
+
+/**
+ * Convert period of time (in minutes) to height (pixels in y dim).
+ *
+ * @param {*} minutesDelta period of time in minutes
+ * @param {*} verticalResolution resolution in minutes
+ * @returns pixels
+ */
+export const minutesToHeight = (minutesDelta, verticalResolution) => {
+  'worklet';
```

Comment:
could you explain this line?

Reply:
This line declares the function as a [rn-reanimated worklet](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/worklets). In summary, it means its code can be run in the UI thread --> more smooth animations

(I added some more context in the file docstring)

Reply:
I just read from the doc, it says to run on UI thread, we must put the function to `runOnUI` function. Am I missing anything?

Reply:
Also run on the UI if is called from a reanimated hook, like `useDerivedValue`, `useAnimatedStyle`, etc. (I just know this from experience, not from the docs: if I try to use a non-`'worklet'` function with a hook --> throws error asking for a worklet)

Reply:
I just noticed this:

* `minutesInDayToTop()` is called from a `useDerivedValue()` function --> needs to be a worklet
* `minutesInHeight()` does not need to be a worklet

I removed the worklet from the latter